kamon-logback-stackdriver/src/main/scala/kamon/stackdriver/StackdriverMarker.scala

```diff
@@ -0,0 +1,80 @@
+package kamon.stackdriver
+
+import java.util
+
+import kamon.stackdriver.StackdriverMarker._
+import org.slf4j.Marker
+
+import scala.collection.concurrent.TrieMap
+import scala.collection.JavaConverters._
+
+abstract class StackdriverMarker(implicit enc: LogValue.Writer) extends Marker {
+  private val markers = TrieMap.empty[String, Marker]
+
+  def values: Map[String, LogValue] =
+    markers.collect {
+      case (name, marker: StackdriverMarker) =>
```

Comment:
I think you should return here all? I'm not sure if `this` should be also returned from iterator.